packages/fuels-programs/src/contract.rs

```diff
@@ -615,9 +617,9 @@ where
     /// let params = CallParameters { amount: 1, asset_id: BASE_ASSET_ID };
     /// my_contract_instance.my_method(...).call_params(params).call()
     /// ```
-    pub fn call_params(mut self, params: CallParameters) -> Result<Self, Error> {
+    pub fn call_params(mut self, params: CallParameters) -> Result<Self> {
         if !self.is_payable() && params.amount > 0 {
-            return Err(Error::InvalidCallParameters());
+            return Err(Error::AssetsForwardedToNonPayableMethod());
```

Comment:
I'd declare this error as `Error::AssetsForwardedToNonPayableMethod` to avoid having to append parentheses whenever you're creating it.

When you add the parentheses the variant type is no longer Unit, but an empty tuple.

Also you then have to add the parentheses when matching as well: 
```rust
        let some_error = Error::AssetsForwardedToNonPayableMethod();
        match some_error {
            Error::AssetsForwardedToNonPayableMethod() => {}
            _ => {}
        }
```